cot-macros/src/lib.rs

```diff
@@ -192,3 +192,82 @@ pub(crate) fn cot_ident() -> proc_macro2::TokenStream {
         }
     }
 }
+
+#[proc_macro_derive(FromRequestParts)]
+pub fn derive_from_request_parts(input: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(input as syn::DeriveInput);
+    impl_from_request_parts_for_struct(&ast).into()
+}
+
+fn impl_from_request_parts_for_struct(ast: &syn::DeriveInput) -> proc_macro2::TokenStream {
```

Comment:
Please move this function to a separate file, similarly to other proc/derive macro `impl` functions.